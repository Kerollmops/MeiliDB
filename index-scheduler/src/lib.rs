mod autobatcher;
mod batch;
pub mod error;
mod index_mapper;
mod index_scheduler;
pub mod task;
mod utils;

pub type Result<T> = std::result::Result<T, Error>;
pub type TaskId = u32;

pub use crate::index_scheduler::{IndexScheduler, Query};
pub use error::Error;
pub use task::{Kind, KindWithContent, Status, TaskView};

#[cfg(test)]
mod tests {
    #[macro_export]
    macro_rules! assert_smol_debug_snapshot {
        ($value:expr, @$snapshot:literal) => {{
            let value = format!("{:?}", $value);
            insta::assert_snapshot!(value, stringify!($value), @$snapshot);
        }};
        ($name:expr, $value:expr) => {{
            let value = format!("{:?}", $value);
            insta::assert_snapshot!(Some($name), value, stringify!($value));
        }};
        ($value:expr) => {{
            let value = format!("{:?}", $value);
            insta::assert_snapshot!($crate::_macro_support::AutoName, value, stringify!($value));
        }};
    }

    #[test]
    fn simple_new() {
        crate::IndexScheduler::test();
    }
}
