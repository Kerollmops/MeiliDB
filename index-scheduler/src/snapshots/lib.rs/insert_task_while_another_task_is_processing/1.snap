---
source: index-scheduler/src/lib.rs
---
### Autobatching Enabled = true
### Processing Tasks:
[0,]
----------------------------------------------------------------------
### All Tasks:
0 {uid: 0, status: enqueued, kind: Snapshot}
1 {uid: 1, status: enqueued, kind: Snapshot}
2 {uid: 2, status: enqueued, kind: IndexDeletion { index_uid: "doggos" }}
----------------------------------------------------------------------
### Status:
enqueued [0,1,2,]
----------------------------------------------------------------------
### Kind:
"indexDeletion" [2,]
"snapshot" [0,1,]
----------------------------------------------------------------------
### Index Tasks:
doggos [2,]
----------------------------------------------------------------------
### Index Mapper:
[]
----------------------------------------------------------------------
### File Store:

----------------------------------------------------------------------

